### PR TITLE
Move TestInfrastructure work out of DynamoCoreWpf

### PR DESCRIPTION
**Pull request by aosyatnik**
### Purpose

Link to YouTrack:
[MAGN-5737](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-5737) Move TestInfrastructure work out of DynamoCoreWpf

Mutators moved to `MutationTestInfrastructure` project, that can be found in `Tests\Core` folder.

There was one confusing thing: every `Mutator` uses `DynamoViewModel`, so `MutationTestInfrastructure` needs `DynamoCoreWpf` reference. But `DynamoViewModel` also uses member of `MutationTestInfrastructure` in `OnModelCommandCompleted`. So, here we have circular dependency. That is really incorrect.

So, I moved `MutateTestCommand` out of `DynamoViewModel` and created `DynamoViewModelMutation`, that should be used just in Mutation tests.

----------------------------------------
Another confusing thing. I was looking for some test, that I can run to be sure, that everything works correctly. But I couldn't find such one. Maybe, `TestMutation` in `DynamoCoreTests`. But, I'm not sure.

But if `TestMutation` is correct test for check. Then I have 94 passed tests and 2 failed.
image

The reason of these tests fail is : 
image

### Declarations

- [x] The code base is in a better state after this PR
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@lukechurch

### CODE REVIEW

**Comment by lukechurch** [comment]
@aosyatnik In order to test, please start a debug build. Open a graph, set run to manual, and run 'debug | mutation test'

It'd be great if you could do this and let me know if it behaves itself

File: src/DynamoCoreWpf/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -30,3 +30,4 @@
 [assembly: InternalsVisibleTo("DynamoMSOfficeTests")]
 [assembly: InternalsVisibleTo("CoreNodeModelsWpf")]
 [assembly: InternalsVisibleTo("DynamoSandbox")]
+[assembly: InternalsVisibleTo("MutationTestInfrastructure")]
```
Comment (lukechurch) [inline src/DynamoCoreWpf/Properties/AssemblyInfo.cs]: @pboyer I'm not fantastically happy about using this mechanism -- but this is a test tool so it does have some plausibility.

We could open it up as an API via the extension but I'm not sure we have the bandwidth to add the APIs with the design attention they'd need right now. What are you thoughts here?

File: test/MutationTestInfrastructure/DynamoViewModelMutation.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿using Dynamo.Models;
+using Dynamo.ViewModels;
+using System;
+
+namespace MutationTestInfrastructure
```
Comment (lukechurch) [inline test/MutationTestInfrastructure/DynamoViewModelMutation.cs]: This doesn't really below in test to be honest. It's a utility class more than anything else. Could we move it to tools?

**Comment by aosyatnik** [comment]
Close this. Create the newest version here: https://github.com/DynamoDS/Dynamo/pull/5128

**Pull request closed without merging**